specs/driver_spec.rb

```diff
@@ -1,104 +1,175 @@
+      end
+
+      it "correctly calculates the average rating" do
+        trip2 = RideShare::Trip.new(id: 8, driver: @driver, passenger: nil,
+                                    start_time: Time.parse("2016-08-08"),
+                                    end_time: Time.parse("2016-08-09"),
+                                    rating: 1)
+        @driver.add_driven_trip(trip2)
+
+        expect(@driver.average_rating).must_be_close_to (5.0 + 1.0) / 2.0, 0.01
       end
 
-      expect(@driver.id).must_be_kind_of Integer
-      expect(@driver.name).must_be_kind_of String
-      expect(@driver.vehicle_id).must_be_kind_of String
-      expect(@driver.status).must_be_kind_of Symbol
+
     end
-  end
 
-  describe "add_driven_trip method" do
+  describe "total_revenue" do
     before do
-      pass = RideShare::User.new(id: 1, name: "Ada", phone: "412-432-7640")
-      @driver = RideShare::Driver.new(id: 3, name: "Lovelace", vin: "12345678912345678")
-      @trip = RideShare::Trip.new({id: 8, driver: @driver, passenger: pass, date: "2016-08-08", rating: 5})
+      @driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV",
+                                      vin: "1C9EVBRM0YBC564DZ", status: :AVAILABLE)
+      trip_1 = RideShare::Trip.new(id: 8, driver: @driver, passenger: nil,
+                                 start_time: Time.parse("2016-08-08"),
+                                 end_time: Time.parse("2016-08-08"), rating: 5, cost: 100)
+
+      trip_2 = RideShare::Trip.new(id: 8, driver: @driver, passenger: nil,start_time: Time.parse("2016-08-08"),
+      end_time: Time.parse("2016-08-08"), rating: 5, cost: 10)
+
+      @driver.add_driven_trip(trip_1)
+      @driver.add_driven_trip(trip_2)
+    end
+
+    it 'returns a float' do
+      expect(@driver.total_revenue).must_be_kind_of Float
     end
 
-    it "throws an argument error if trip is not provided" do
-      expect{ @driver.add_trip(1) }.must_raise ArgumentError
+    it 'returns zero for no driven trips' do
+      driver_2 = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV",vin: "1C9EVBRM0YBC564DZ", status: :AVAILABLE)
+
+      expect(driver_2.total_revenue).must_equal 0
     end
 
-    it "increases the trip count by one" do
-      previous = @driver.trips.length
-      @driver.add_trip(@trip)
-      expect(@driver.trips.length).must_equal previous + 1
+    it 'correctly calculates the total' do
+      expect(@driver.total_revenue).must_equal 85.36
+    end
+
+    it 'returns a floating point with correct money format' do
+      trip_3 = RideShare::Trip.new(id: 8, driver: @driver, passenger: nil,start_time: Time.parse("2016-08-08"),
+      end_time: Time.parse("2016-08-08"), rating: 5, cost: 66.66)
+
+      @driver.add_driven_trip(trip_3)
+
+      expect(@driver.total_revenue).must_equal 137.37
     end
   end
 
-  describe "average_rating method" do
+  describe "net_expenditures" do
     before do
       @driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV",
```

Comment:
You're missing some interesting cases here. What if a driver has

- No driven trips
- No trips as a passenger
- An incomplete trip as either a driver or passenger
- Made more money than they've spent